src/python/WMCore/WMSpec/WMTask.py

```diff
@@ -926,10 +927,19 @@ def setSubscriptionInformation(self, custodialSites = None, nonCustodialSites =
             outputDataset = entry["outputDataset"]
             outputModule = entry["outputModule"]
 
-            primDs = outputDataset.split('/')[1]
-            tier = outputDataset.split('/')[3]
+            dsSplit = outputDataset.split('/')
+
+            primDs = dsSplit[1]
+
+            procDsSplit = dsSplit[2].split('-')
+            skim = ( len(procDsSplit) == 4 )
```

Comment:
That's a failure in pylint IMO, I find the former easier to read (as it more clearly separates the assignment and comparison operator.

Two flags because one is for "should I care if this is a skim or not", the second is for "now that I care, do you want me to configure for a skim or a non-skim".

And yes, the two flags propagate all the way down from the top, that's what most of this PR does. Tier0 calls setSubscriptionInfo on the workload. To be honest, I don't see a need for WMAgent to ever use this, your workflows are much simpler and don't need this.